### ISSUE TITLE
Keyboard access and focus for accessible use of clickable badges

### ISSUE THREAD

**Issue by allisonsuarez** [amundsen-io/amundsen#649]
The clickable badges should be keyboard accessible and focusable.

Based on conversation from [previous PR](https://github.com/lyft/amundsenfrontendlibrary/pull/529#discussion_r463397108) 

## Expected Behavior or Use Case
A user can use tab to reach and click on badges.

## Service or Ingestion ETL
amundsenfrontendlibrary

## Possible Implementation
Decouple Flag and ClickableBadge components to resolve accessibility issue and maintain single responsibility for the Flag component.

## Context
Make Amundsen better for users with accessibility needs.
